ca/certificates.go

```diff
@@ -410,8 +423,22 @@ func NewRootCA(certBytes, keyBytes []byte, certExpiry time.Duration) (RootCA, er
 
 func ensureCertKeyMatch(cert *x509.Certificate, key crypto.PublicKey) error {
 	switch certPub := cert.PublicKey.(type) {
-	// TODO: Handle RSA keys.
+	case *rsa.PublicKey:
+		if certPub.N.BitLen() < 2048 {
+			return errors.New("RSA keys must be >= 2048 bits")
+		}
+		rsaKey, ok := key.(*rsa.PublicKey)
+		if ok && certPub.E == rsaKey.E && certPub.N.Cmp(rsaKey.N) == 0 {
```

Comment:
I can definitely to make sure E != 1, but I'm not sure stuff like the golang APIs or openssl will happily generate a key with such.  I mostly want to make sure really old keys aren't being used or something.